test/testdata/rbs/signatures_defs.rb

```diff
@@ -40,13 +40,47 @@ def parse_error4(p1, p2); end # error: The method `parse_error4` does not have a
 def sig_mismatch1(p1, p2); end
 #                     ^^ error: Malformed `sig`. Type not specified for argument `p2`
 
-#: (foo: P1) -> void # error: Unknown argument name `foo`
+#: (foo: P1) -> void
+#   ^^^ error: Argument kind mismatch for `p1`, expected `positional`, got `keyword`
+#   ^^^ error: Unknown argument name `foo`
 def sig_mismatch2(p1); end
 #                 ^^ error: Malformed `sig`. Type not specified for argument `p1`
 
 #: (P1, P2, P3) -> void # error: RBS signature has more parameters than in the method definition
 def sig_mismatch3; end # error: The method `sig_mismatch3` does not have a `sig`
 
+#: (?Integer) -> void
+#    ^^^^^^^ error: Argument kind mismatch for `p`, expected `positional`, got `optional positional`
+def sig_mismatch4(p); end
+
+#: (Integer) -> void
+#   ^^^^^^^ error: Argument kind mismatch for `p`, expected `optional positional`, got `positional`
+def sig_mismatch5(p = 42); end
+
+#: (Integer) -> void
+#   ^^^^^^^ error: Argument kind mismatch for `p`, expected `keyword`, got `positional`
+def sig_mismatch6(p:); end
+
+#: (Integer) -> void
+#   ^^^^^^^ error: Argument kind mismatch for `p`, expected `rest positional`, got `positional`
+def sig_mismatch7(*p); end
+
+#: (?p: Integer) -> void
+#    ^ error: Argument kind mismatch for `p`, expected `keyword`, got `optional keyword`
+def sig_mismatch8(p:); end
+
+#: (p: Integer) -> void
+#   ^ error: Argument kind mismatch for `p`, expected `optional keyword`, got `keyword`
+def sig_mismatch9(p: 42); end
+
+#: (*Integer) -> void
+#    ^^^^^^^ error: Argument kind mismatch for `p`, expected `rest keyword`, got `rest positional`
+def sig_mismatch10(**p); end
+
+#: (^() -> void) -> void
+#   ^^^^^^^^^^^ error: Argument kind mismatch for `blk`, expected `block`, got `positional`
+def sig_mismatch11(&blk); end
+
```

Comment:
Could you add a test case where the mismatch is in one of the args to the block?